kubetest/gke.go

```diff
@@ -462,12 +458,20 @@ export KUBE_NODE_OS_DISTRIBUTION='%[3]s'
 		log.Printf("error while ensuring firewall rule: %v", err)
 	}
 
-	return control.FinishRunning(exec.Command("bash", "-c", fmt.Sprintf(gkeLogDumpTemplate,
-		g.project,
-		g.zone,
-		os.Getenv("NODE_OS_DISTRIBUTION"),
-		strings.Join(filters, " OR "),
-		dumpCmd)))
+	var lastError error
+	for zone, filters := range perZoneFilters {
+		err := control.FinishRunning(exec.Command("bash", "-c", fmt.Sprintf(gkeLogDumpTemplate,
+			g.project,
+			zone,
+			os.Getenv("NODE_OS_DISTRIBUTION"),
+			strings.Join(filters, " OR "),
+			dumpCmd)))
+		if err != nil {
+			log.Printf("error while dumping logs in %s zone: %v", zone, err)
```

Comment:
Let's not print and return the same error

Reply:
I wanted to print it, as this error message is actually discarded in the calling function:
https://github.com/kubernetes/test-infra/blob/master/kubetest/e2e.go#L122

Reply:
Nevermind. That call was only for cases where test failed during up. In other places error is handled.

Reply:
Done